storagenode/storagenodedb/pieceexpiration.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package storagenodedb
+
+import (
+	"context"
+	"time"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storagenode/pieces"
+)
+
+type pieceExpirationDB struct {
+	*InfoDB
+}
+
+// PieceExpirationDB returns database for storing piece expiration data
+func (db *DB) PieceExpirationDB() pieces.PieceExpirationDB { return db.info.PieceExpirationDB() }
+
+// PieceExpirationDB returns database for storing piece expiration data
+func (db *InfoDB) PieceExpirationDB() pieces.PieceExpirationDB { return &db.pieceExpirationDB }
+
+// GetExpired gets piece IDs that expire or have expired before the given time
+func (db *pieceExpirationDB) GetExpired(ctx context.Context, expiredAt time.Time, limit int64) (expiredPieceIDs []pieces.ExpiredInfo, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	rows, err := db.db.QueryContext(ctx, `
+		SELECT satellite_id, piece_id
+			FROM piece_expirations
+			WHERE piece_expiration < ?
+				AND ((deletion_failed_at IS NULL) OR deletion_failed_at <> ?)
+			LIMIT ?
+	`, expiredAt.UTC(), expiredAt.UTC(), limit)
+	if err != nil {
+		return nil, ErrInfo.Wrap(err)
+	}
+	defer func() { err = errs.Combine(err, rows.Close()) }()
+
+	for rows.Next() {
+		var satelliteID storj.NodeID
+		var pieceID storj.PieceID
+		err = rows.Scan(&satelliteID, &pieceID)
+		if err != nil {
+			return nil, ErrInfo.Wrap(err)
+		}
+		expiredPieceIDs = append(expiredPieceIDs, pieces.ExpiredInfo{
+			SatelliteID: satelliteID,
+			PieceID:     pieceID,
+			InPieceInfo: false,
+		})
+	}
+	return expiredPieceIDs, nil
+}
+
+// SetExpiration sets an expiration time for the given piece ID on the given satellite
+func (db *pieceExpirationDB) SetExpiration(ctx context.Context, satellite storj.NodeID, pieceID storj.PieceID, expiresAt time.Time) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	_, err = db.db.ExecContext(ctx, `
+		INSERT INTO piece_expirations(satellite_id, piece_id, piece_expiration)
+			VALUES (?,?,?)
+	`, satellite, pieceID, expiresAt.UTC())
+	return ErrInfo.Wrap(err)
+}
+
+// DeleteExpiration removes an expiration record for the given piece ID on the given satellite
```

Comment:
I'm finding it fairly difficult to care about periods at the end of comments, especially when so many of the existing comments don't have them. If periods are a matter of great concern for us, let's get all existing code into shape first, and add it to the linter.

Reply:
Yes, it's true that some of them are missed, but I don't know if it's because we didn't care before and now we do or because they haven't been caught in reviews.

I agree that the linter should report them; however, I don't think that now there will be a task assign to somebody for doing so when we have tons of other important things to do, so I think that if they are caught in a review is kind of doing our best effort for aligning the shape of the sources.

A few days ago I asked for adding them or removing them from one new go file because some methods had and others didn't only for keeping it consistent in the same file, but I acknowledged that our conventions point to use the Go code review comments and they refer to use _period_ at then of them.

By the way, I would have posted the suggestion but it turns that I've realized about the feature later in the review and although I scrolled up to find some of these that I did before that I missed this one.